setup/upgrade_major.rst

```diff
@@ -186,3 +192,123 @@ Next, use Composer to download new versions of the libraries:
+
+Symfony 6 will come with native PHP return types to (almost all) methods.
+It is important to add the native PHP return types to your classes first.
+Otherwise, you will get incompatible declaration errors when upgrading to
+Symfony 6.0.
+
+When debug mode is enabled (typically in the dev and test environment),
+Symfony will trigger deprecations for every incompatible method
+declaration. For instance, the ``UserInterface::getRoles()`` method will
+have an ``array`` return type in Symfony 6. In order to be compatible, you
+must add the same return type to the ``getRoles()`` method while using
+Symfony 5.4.
+
+To help with this, Symfony provides a script that can add these native
+return types automatically for you. Make sure you installed the
+``symfony/error-handler`` component. When installed, generate a complete
+class map using Composer and run the script to iterate over the class map
+and fix any incompatible method:
+
+.. code-block:: terminal
+
+    # "-o" is important! This forces Composer to find all classes
+    $ composer dump-autoload -o
+
+    # patch all incompatible method declarations
+    $ ./vendor/bin/patch-type-declarations
+
+.. tip::
+
+    This feature is not limited to Symfony packages. It will also help you
+    add types and prepare for other dependencies in your project.
+
+The behavior of this script can be modified using the
+``SYMFONY_PATCH_TYPE_DECLARATIONS`` env var. The value of this env var is
+url-encoded (e.g. ``param1=value2&param2=value2``), the following
+parameters are available:
+
+``force``
+    Enables fixing return types, the value must be one of:
+
+    * ``2`` to add all possible return types (default, recommended for applications);
+    * ``1`` to add return types only to tests, final, internal or private methods;
+    * ``phpdoc`` to only add ``@return`` docblock annotations to the incompatible
+      methods, or ``#[ReturnTypeWillChange]`` if it's triggered by the PHP engine.
+
+``php``
+    The target version of PHP - e.g. ``7.1`` doesn't generate "object"
+    types (which were introduced in 7.2). This defaults to the PHP version
+    used when running the script.
+
+``deprecations``
+    Set to ``0`` to disable deprecations. Otherwise, a deprecation notice
+    when a child class misses a return type while the parent declares an
+    ``@return`` annotation (defaults to ``1``).
+
+If there are specific files that should be ignored, you can set the
+``SYMFONY_PATCH_TYPE_EXCLUDE`` env var to a regex. This regex will be
+matched to the full path to the class and each matching path will be
+ignored (e.g. ``SYMFONY_PATCH_TYPE_EXCLUDE="/tests\/Fixtures\//"``).
+Classes in the ``vendor/`` directory are always ignored.
+
+.. tip::
+
+    The script does not care about code style. Run your code style fixer,
+    or `PHP CS Fixer`_ with the ``phpdoc_trim_consecutive_blank_line_separation``
+    and ``no_superfluous_phpdoc_tags`` rules, after patching the types.
```

Comment:
there might also be the one that orders `use` statements if you want one more :)